app/src/main/java/com/cornellappdev/coffee_chats_android/SchedulingActivity.kt

```diff
@@ -0,0 +1,101 @@
+        setContentView(R.layout.activity_scheduling)
+
+        val timeAdapter = TimeOptionAdapter(this, times)
+        time_gridview.adapter = timeAdapter
+        var dayAdapter = DayAdapter(this, days)
+        day_selection.adapter = dayAdapter
+
+        day_header.text = "Every Sunday"        // Sunday by default
+        for (day in daysFullName) {
+            selectedTimes[day] = mutableListOf<String>()
+        }
+
+        back_button.setOnClickListener { finish() }
+
+        var previousDot: ImageView? = null
+        day_selection.onItemClickListener = OnItemClickListener { parent, v, position, id ->
+//            val daySelectedIndex = dayAdapter.selectedPositions.indexOf(position)
+            val daySelectedView = day_selection.getChildAt(position) as ConstraintLayout
+            val daySelectedTextView = daySelectedView.getChildAt(0) as TextView
+            val daySelectedDot = daySelectedView.getChildAt(1) as ImageView
+            val sundaySelectedView = day_selection.getChildAt(0) as ConstraintLayout
+            val sundaySelectedDot = sundaySelectedView.getChildAt(1) as ImageView
+            if (previousDot == null) sundaySelectedDot.visibility = View.INVISIBLE
+            if (previousDot != null) previousDot!!.visibility = View.INVISIBLE
+            daySelectedDot.visibility = View.VISIBLE
+            previousDot = daySelectedDot
+            day_header.text = "Every " + daysFullName[position]
+            currDay = daysFullName[position]
+            // update the time gridview to reflect selected time slots
+            for (i in times.indices) {
+                val timeView = time_gridview.getChildAt(i) as LinearLayout
+                val timeTextView = timeView.getChildAt(0) as TextView
+
+                if (selectedTimes[currDay]!!.contains(times[i])) {
+                    timeTextView.background = getDrawable(R.drawable.selected_rounded_time_option)
+                } else {
+                    timeTextView.background = getDrawable(R.drawable.unselected_rounded_time_option)
+                }
+            }
+        }
+
+        time_gridview.onItemClickListener = OnItemClickListener { parent, v, position, id ->
+            val timeSelectedView = time_gridview.getChildAt(position) as LinearLayout
+            val timeSelectedTextView = timeSelectedView.getChildAt(0) as TextView
+            val timeSelectedIndex = selectedTimes[currDay]!!.indexOf(timeSelectedTextView.text.toString())
+            if (timeSelectedIndex > -1) {
+                selectedTimes[currDay]!!.remove(timeSelectedTextView.text.toString())
+                timeSelectedTextView.background = getDrawable(R.drawable.unselected_rounded_time_option)
+                // change the day button to white if no time is selected for current day
+                if (selectedTimes[currDay]!!.size == 0) {
+                    selectedDays.remove(currDay)
+                    val currDayIndex = daysFullName.indexOf(currDay)
```

Comment:
ideally the logistical code is separated from the ui code thru a method call, where the logic code can live in the presenter and the ui code lives in the activity. combining both makes the file more difficult to read

Reply:
Great idea. I'll add that as an issue and tackle it later